lib/sycamore/sycamore/transforms/merge_elements.py

```diff
@@ -135,7 +135,151 @@ def merge(self, elt1: Element, elt2: Element) -> Element:
+        if "page_number" not in element1.properties or "page_number" not in element2.properties or element1.type == None or element2.type == None:
+            return False
+
+        # DO NOT MERGE across pages
+        if not self.merge_across_pages and element1.properties["page_number"] != element2.properties["page_number"]:
+            return False
+
+        if element1.data["token_count"] + 1 + element2.data["token_count"] > self.max_tokens:
+            return False
+        
+        # MERGE adjacent 'text' elements (but not across pages - see above)
+        if ((element1 != None) and (element1.type == 'Text') and 
+            (element2 != None) and (element2.type == 'Text')):
+            return True
+        
+        # MERGE 'Section-header' + 'table' 
+        if ((element1 != None) and (element1.type == 'Section-header') and 
+            (element2 != None) and (element2.type == 'table')):
+            return True
+        
+        # Merge 'image' + 'text'* until next 'image' or 'Section-header' 
+        if ((element1 != None) and (element1.type == 'Image' or element1.type == 'Image+Text') and 
+            (element2 != None) and (element2.type == 'Text')):
+            return True
+        return False
+                
+    def merge(self, elt1: Element, elt2: Element) -> Element:
+        """Merge two elements; the new element's fields will be set as:
+            - type: "Section"
+            - binary_representation: elt1.binary_representation + elt2.binary_representation
+            - text_representation: elt1.text_representation + elt2.text_representation
+            - bbox: the minimal bbox that contains both elt1's and elt2's bboxes
+            - properties: elt1's properties + any of elt2's properties that are not in elt1
+            note: if elt1 and elt2 have different values for the same property, we take elt1's value
+            note: if any input field is None we take the other element's field without merge logic
+
+        Args:
+            element1 (Tuple[Element, int]): the first element (and number of tokens in it)
+            element2 (Tuple[Element, int]): the second element (and number of tokens in it)
+
+        Returns:
+            Tuple[Element, int]: a new merged element from the inputs (and number of tokens in it)
+        """
+
+        tok1 = elt1.data["token_count"]
+        tok2 = elt2.data["token_count"]
+        new_elt = Element()
+        # 'text' + 'text' = 'text'
+        # 'image' + 'text' = 'image+text'
+        # 'image+text' + 'text' = 'image+text'
+        # 'Section-header' + 'table' = 'Section-header+table'
+
+        if (elt1.type == "Image" or elt1.type == "Image+Text") and elt2.type == "Text":
+            new_elt.type = "Image+Text"
+        elif elt1.type == "Text" and elt2.type == "Text":
+            new_elt.type = "Text"
+        elif elt1.type == "Section-header" and elt2.type == "table":
+            new_elt.type = "Section-header+table"
```

Comment:
I'm wondering a little bit about changing the element types here and using a generic Element. It means we lose all of the utility methods in TableElement/Table an ImageElement. Another option might be to add a "title" field to a TableElement (it already has a caption) and add some description field to an ImageElement. 

I won't be a stickler here -- this will obviously work for a lot of use cases, just trying to think through how element types should work going forward.

Reply:
We should at least keep the table data accessible when merging elements. I'd also vote for Section-header+Table just being an augmented Table object. We can update the text_representation and store the title as a field

Reply:
Agreed that more though is needed in how the new element types are named and built. 
What I wanted to show with these names is that the merged elements are no longer just a Table, or an Image.